bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/AdaptiveImageServlet.java

```diff
@@ -185,8 +185,10 @@ protected void doGet(@Nonnull SlingHttpServletRequest request, @Nonnull SlingHtt
                 lastModifiedEpoch = assetLastModifiedEpoch;
             }
         }
+        
         long requestLastModifiedSuffix = getRequestLastModifiedSuffix(suffix);
-        if (requestLastModifiedSuffix >= 0 && requestLastModifiedSuffix != lastModifiedEpoch) {
+        if (requestLastModifiedSuffix >= 0 && requestLastModifiedSuffix != lastModifiedEpoch 
```

Comment:
the redirect is always necessary IMHO, the only thing that changes is from where to extract the lastModified timestamp. This should be before the first dot of the suffix! That should be compliant with both the old and the new URL format.

Reply:
The timestamp doesn't exist before the first dot of the suffix in case of a component not as Image resource type. Hence I have used a regex to find it in the suffix string. This regex is checking for a 13 digit numeric string which would be the case of a valid UTC timestamp string converted in milliseconds. 
Redirection logic is now completely untouched!